src/components/RandomPresent/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { LeftStep, useTypedTranslation, useInterval, RangePrice, type StylePropType, useScreenWidth } from "@shared";
+import { useEffect, useState } from "react";
+import { useGetRandomProductsQuery } from "@src/app/api/products";
+import { getLeft, getSize, getDirection, getStyle, getPresent } from "./helpers";
+import { useAppDispatch, useAppSelector } from "@src/app/store";
+import { setPermition } from "@src/app/store/slices/randomRange";
+
+export default function RandomPresent (): JSX.Element {
+  const randomRange = useAppSelector((state) => state.randomRange);
+  const { from, to } = randomRange;
+  const { data, isLoading, error } = useGetRandomProductsQuery({ from, to });
```

Comment:
Here as I can see u try to get dynamically images for random present based on price range and save this range in store.
I think it is not good approach to do this.
First of all each time user change slider we do new request - it is bad. We need to get images once - when user stop to edit price range.
Next, we do not need to update store on each change. The same as for requests - update it only when user stop changing range.

Just open DevTools -> Network and see a lot of requests. Now assume our user has slow internet.